src/oc/auth/api/users.clj

```diff
@@ -64,9 +64,11 @@
 
 (defn- valid-user-update? [conn user-props user-id]
   (if-let [user (user-res/get-user conn user-id)]
-    (let [new-password (:password user-props)
-          updated-user (merge user (user-res/ignore-props user-props))]
-      (if (lib-schema/valid? user-res/User updated-user)
+    (let [current-password (:current-password user-props)
+          new-password (:password user-props)
+          updated-user (merge user (user-res/ignore-props (dissoc user-props :current-password)))]
+      (if (and (lib-schema/valid? user-res/User updated-user)
+               (user-res/password-match? current-password (:password-hash user)))
```

Comment:
This won't work when user isn't updating their password since it's not sent and current password will be compared to a hash of nil.